### PR TITLE
Add ability to change copy behavior

### LINKED ISSUE #1064: [Feature Request] Copy on first tap when using tap to reveal
I would like a feature where it already copies the token to the clipboard on the same tap when using the "Tap to reveal" and "Copy tokens when tapped" option.
My use case is that I always copy tokens from mobile and KDE Connect syncs my clipboard, and then I can paste it directly on my desktop.

### PR DESCRIPTION
**Pull request by michaelschattgen**
In the past we've received a couple different reports (see #1064, #690 and #1120) on how the copy to tap should work in combination with our "tap to reveal" feature. This pull request allows the user to manually set the copy behavior to their liking using the following options:

image

I've also added migration logic to convert the old "copy on tap" boolean to "single tap" to make the update a bit smoother for our users.

This way I hope to please everyone since I've tried to consider everyone's wishes and various situations. Please let me know if you think I've missed anything.

Closes #1064

### CODE REVIEW

File: app/src/main/java/com/beemdevelopment/aegis/Preferences.java [hunk 1]
```diff
@@ -45,6 +45,23 @@ public Preferences(Context context) {
         if (getPasswordReminderTimestamp().getTime() == 0) {
             resetPasswordReminderTimestamp();
         }
+
+        migratePreferences();
+    }
+
+    public void migratePreferences()
+    {
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/Preferences.java:53]: Style nit.

File: app/src/main/java/com/beemdevelopment/aegis/ui/views/EntryAdapter.java [hunk 2]
```diff
@@ -437,24 +439,40 @@ public void onClick(View v) {
...
                                 handled = true;
                             } else {
                                 focusEntry(entry, _tapToRevealTime);
                             }
-                        } else {
-                            _copiedEntry = null;
+                        }
+
+                        if (!copiedThisClick)
+                        {
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/views/EntryAdapter.java:474]: Style nit.

File: app/src/main/java/com/beemdevelopment/aegis/ui/views/EntryListView.java [hunk 3]
```diff
@@ -173,8 +174,9 @@ public void setIsLongPressDragEnabled(boolean enabled) {
         _touchCallback.setIsLongPressDragEnabled(enabled && _adapter.isDragAndDropAllowed());
     }
 
-    public void setIsCopyOnTapEnabled(boolean enabled) {
-       _adapter.setIsCopyOnTapEnabled(enabled);
+    public void setCopyBehavior(CopyBehavior copyBehavior)
+    {
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/views/EntryListView.java:178]: Style nit.

**Review by alexbakker — CHANGES_REQUESTED** [review]
Overall this looks good to me, but I did notice one small bug:
- Set up Aegis to copy on single tap and have tap to reveal / tap to highlight enabled
- Tap an entry and notice that it correctly copies the code to the clipboard
- Tap the entry again and notice that the entry doesn't get unhighlighted / the code doesn't get hidden.

**Comment by michaelschattgen** [comment]
Fixed styling issues and the bug you described. Also merged master into this feature branch,

**Review by alexbakker — APPROVED** [review]
LGTM!

**Comment by D3SOX** [comment]
I don't want to be rude, but is a new release already planned where this change will be included?

**Comment by michaelschattgen** [comment]
It's not rude, don't worry. We don't like to give ETA's since we work on this in our spare time, that's why we also don't plan ahead when it comes to releases. However we aim to release a beta version this week/next week and hope to release to production soon after depending on the stability.

**Pull request merged**